tools/docker/Dockerfile

```diff
@@ -1,47 +1,117 @@
-FROM ubuntu:18.04
-LABEL maintainer="Edoardo Morassutto <edoardo.morassutto@gmail.com>"
+FROM ubuntu:24.04
 
-ARG UID=1000
-ARG GID=1000
+# Example invocation of docker build
+# $ docker build \
+#     --build-arg TM_VERSION=X.Y.Z \
+#     --build-arg VCS_REF="$(git rev-parse HEAD)" \
+#     --build-arg BUILD_DATE="$(date -u +%Y-%m-%dT%H:%M:%SZ)" \
+#     -t <yourrepo>/task-maker-rust:X.Y.Z
+#       .
+
+ARG TM_VERSION        # task-maker-rust version (required)
+ARG VCS_REF           # git commit SHA for v${TM_VERSION}
+ARG BUILD_DATE        # e.g. 2025-09-04T12:34:56Z (RFC 3339 UTC)
+ARG IMAGE_URL
+ARG DOCS_URL='https://github.com/olimpiadi-informatica/task-maker-rust#readme'
+ARG VENDOR='Olimpiadi Italiane di Informatica'
+ARG BASE_NAME='ubuntu:24.04'
+ARG BASE_DIGEST='sha256:9cbed754112939e914291337b5e554b07ad7c392491dba6daf25eef1332a22e8'
+
+LABEL \
+  org.opencontainers.image.title="task-maker-rust" \
+  org.opencontainers.image.description="task-maker-rust server and worker to build programming tasks for CMS." \
+  org.opencontainers.image.version="${TM_VERSION}" \
+  org.opencontainers.image.revision="${VCS_REF}" \
+  org.opencontainers.image.created="${BUILD_DATE}" \
+  org.opencontainers.image.url="${IMAGE_URL}" \
+  org.opencontainers.image.documentation="${DOCS_URL}" \
+  org.opencontainers.image.source="https://github.com/olimpiadi-informatica/task-maker-rust" \
+  org.opencontainers.image.authors="task-maker-rust contributors" \
+  org.opencontainers.image.vendor="${VENDOR}" \
+  org.opencontainers.image.licenses="MPL-2.0" \
+  org.opencontainers.image.base.name="${BASE_NAME}" \
+  org.opencontainers.image.base.digest="${BASE_DIGEST}"
+
+ARG TM_UID=1000
+ARG TM_GID=1000
+
+ENV RUST_LOG='info'
+ENV RUST_BACKTRACE=1
+
+# run the following as root
+USER root
 
 # install dependencies
-RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -yy \
+RUN apt-get update \
+  && DEBIAN_FRONTEND=noninteractive apt-get install -yy \
     asymptote \
     build-essential \
     fpc \
     latexmk \
     libseccomp-dev \
-    python \
-    python-sortedcontainers \
+    libssl-dev \
     python3 \
     python3-sortedcontainers \
     texlive \
     texlive-latex-extra \
     wget \
+    curl \
   && rm -rf /var/lib/apt/lists/*
 
-# task-maker-rust version (required)
-ARG TM_VERSION
+# delete default ubuntu user, this removes also the group
+RUN userdel --remove ubuntu
+
+# create a group and a user called taskmaker, create the home directory
+RUN groupadd ${TM_GID:+-g "${TM_GID}"} taskmaker
```

Comment:
I think we always have called `task-maker` with the dash, can you add it also to the user name?

Reply:
To be fair, in the previous version of the `Dockerfile` the user was called `user` :-D. I will change it to task-maker.